src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,57 @@
 package core.basesyntax;
 
+
+import java.io.*;
+
 public class WorkWithFile {
+    int buySum = 0;
+    int supplySum = 0;
     public void getStatistic(String fromFileName, String toFileName) {
+        readFromFile(fromFileName);
+        calculateReport();
+        writeToFile(toFileName);
+        cleanUp();
+    }
+
+    private void cleanUp() {
+        buySum = 0;
+        supplySum = 0;
+    }
+
+    public void readFromFile(String fromFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))){
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                String type = parts[0];
+                int amount = Integer.parseInt(parts[1]);
+
+                if ("supply".equals(type)) {
+                    supplySum += amount;
+                } else if ("buy".equals(type)) {
+                    buySum += amount;
+                }
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Can't find file ", e);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file ", e);
+        }
+    }
+
+    public int calculateReport() {
+        return supplySum - buySum;
+    }
 
+    public void writeToFile(String toFileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+            StringBuilder builder = new StringBuilder();
+            builder.append("supply,").append(supplySum).append(System.lineSeparator())
+                    .append("buy,").append(buySum).append(System.lineSeparator())
+                    .append("result,").append(calculateReport());
```

Comment:
Using StringBuilder here is good, but you should also consider moving the string literals like 'supply', 'buy', and 'result' into constants to make the code more maintainable.